testsuite/indexer_grpc_local.py

```diff
@@ -172,106 +182,175 @@ def wait_for_indexer_grpc_progress(shell: Shell, client: HttpClient) -> None:
     )
     res = None
     for _ in range(GRPC_PROGRESS_THRESHOLD_SECS // retry_secs):
-        res = shell.run(
-            [
-                "timeout",
-                f"{GRPC_PROGRESS_THRESHOLD_SECS}s",
-                GRPCURL_PATH,
-                "-max-msg-sz",
-                "10000000",
-                "-d",
-                '{ "starting_version": 0 }',
-                "-H",
-                "x-aptos-data-authorization:dummy_token",
-                "-import-path",
-                "crates/aptos-protos/proto",
-                "-proto",
-                "aptos/indexer/v1/raw_data.proto",
-                "-plaintext",
-                GRPC_DATA_SERVICE_URL,
-                "aptos.indexer.v1.RawData/GetTransactions",
-            ],
-        )
-        if (
-            res.exit_code == 124
-        ):  # timeout exits with 124 if it reaches the end of the timeout
+        try:
+            res = context.shell.run(
+                [
+                    GRPCURL_PATH,
+                    "-max-msg-sz",
+                    "10000000",
+                    "-d",
+                    '{ "starting_version": 0 }',
```

Comment:
optional: currently data service supports finite stream, i.e., `transactions_count`. The TimeoutExpired exception may indicate other issues if the finite stream is expected to end already.

Reply:
Okay good to know. I'll leave that as an optional improvement for now.